File is parsed, but no settings applied.

I'm pretty sure to use your plugin as described and can see on Windows that `.editorconfig` is parsed by some Java process. Additionally I see different behaviour of Java in case `root = true` is present in the file vs. when it is not: If it's present only one `.editorconfig` is parsed, if it's not the directory tree is searched upwords for additional files until `C:\.editorconfig`, which is not found in the end.

Running the task `cleanEclipse` as well deletes all your managed files, while the task `eclipse` generates all of those containing the line with the version number only. I've already compared behaviour of both tasks with/without your plugin and there's a difference as well, because some of the `prefs`-files are only created by your plugin, not the one name `eclipse` already.

But whatever I put into `.editorconfig`, besides `root = true` nothing seems to be recognized and I already only copied the examples from you. I don't even get any errors logged, it simply seems that all statements are completely ignored, not with wrong values.

Any idea what could be the problem? Are there any logs available for EditorConfig?

Thanks!

I was able to get EditorConfig to throw an exception because of malformed keys in the section `[*]`:

	09:54:35.263 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] > Failed to apply plugin [id 'org.standardout.eclipseconfig']
	09:54:35.263 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]    > charset  utf-8

So not only the file is parsed, but the section as per your documentation as well properly recognized and the line for the `charset` is read and most likely parsed and stored.

I've finally found the problem: I'm using Windows, your plugin passes native Windows paths using `\` as separator and `EditorConfig` doesn't support that and seems to match no section. Changing `\` to `/` before passing the path to `EditorConfig` works as expected.

	11:01:56.533 [ERROR] [org.gradle.api.Project] C:\[...]\Dummy.java -> C:/[...]/Dummy.java
	11:01:56.535 [ERROR] [system.err] .*/[^/]*
	11:01:56.538 [ERROR] [system.err] .*/[^/]*
	11:01:56.538 [ERROR] [org.gradle.api.Project] [:] -> [charset:utf-8, trim_trailing_whitespace:true, indent_style:space, indent_size:2, tab_width:2]
	11:01:56.543 [ERROR] [system.err] .*/[^/]*

The outputs on `STDERR` are from `EditorConfig` by setting the following on the shell:

> set JAVA_OPTS=-Deditorconfig.debug=true

These are the regular expressions matched against the paths and as you can see, those are using `/` instead of `\`, but there doesn't seem to be any code normalizing paths to use that.

https://github.com/editorconfig/editorconfig-core-java/blob/master/src/main/java/org/editorconfig/core/EditorConfig.java#L224

Should I create a PR for you and report the problem to `EditorConfig` or only the latter? In my opinion `EditorConfig` needs to handle such things more compatible, but you might be able to release sooner.

--- a/src/main/groovy/org/standardout/gradle/econfig/EclipseConfigPlugin.groovy
+++ b/src/main/groovy/org/standardout/gradle/econfig/EclipseConfigPlugin.groovy
@@ -43,10 +43,19 @@ class EclipseConfigPlugin implements Plugin<Project> {
     result
   }
 
-  void applyJavaConfiguration(Project project, EditorConfig editorConfig) {
-    File dummyFile = project.file('Dummy.java')
+  private Map getDummyFileProps(Project project, EditorConfig editorConfig) {
+    File   dummyFile    = project.file('Dummy.java')
+    String dummyFileStr = dummyFile.absolutePath.replaceAll("\\\\", "/")
+    Map    settings     = toMap(editorConfig.getProperties(dummyFileStr))
 
-    def settings = toMap(editorConfig.getProperties(dummyFile.absolutePath))
+    project.logger.debug(".editorconfig read for: ${dummyFile} -> ${dummyFileStr}")
+    project.logger.debug(".editorconfig contents: ${settings}")
+
+    settings
+  }
+
+  void applyJavaConfiguration(Project project, EditorConfig editorConfig) {
+    def settings = getDummyFileProps(project, editorConfig)
 
     // JDT properties
     /* Use of eclipse plugin DSL leads to errors when plugin is used via new plugin mechanism
@@ -155,9 +164,7 @@ class EclipseConfigPlugin implements Plugin<Project> {
   }
 
   void applyCommonConfiguration(Project project, EditorConfig editorConfig) {
-    File dummyFile = project.file('Dummy.java')
-
-    def settings = toMap(editorConfig.getProperties(dummyFile.absolutePath))
+    def settings = getDummyFileProps(project, editorConfig)
 
     // remove properties on clean
     project.tasks.cleanEclipse.doLast {

